Adds QuantizeMultiple, which produces a palette for a slice of images

carbocation opened the pull request:
As noted in #1 , for animated gifs, it is ideal to build a palette that is aware of all of the images that will contribute to the gif. This pull request adds `QuantizeMultiple`, which ingests a slice of images rather than a single image. This preserves `Quantize` in order to prevent breaking the public API. It modifies the private API by changing `buildBucketMultiple` to `buildBucket` and causing it to ingest a slice of images rather than just one. 

Visually/subjectively, this produces nicer appearing images for me than (a) applying Quantize either to every image separately, which can lead to palette jitter between frames, or (b) applying Quantize to just one key image and applying the palette to every image, which can lead to inappropriate coloration in some frames.

carbocation committed:
Adds QuantizeMultiple, which ingests a slice of images and produces
an optimal palette across the set of images.

ericpauley commented:
Thanks for the contribution, this definitely seems useful!

This is perhaps a little pedantic, but could we modify the internal `buildBucketMultiple` api to avoid having to allocate the additional parameter slice? Perhaps make the signature still accept a single image, but also an optional `colorBucket` so that multiple images can be made into one bucket? I doubt either case affects runtime substantially but keeping the magic `allocs` number down is somewhat of a personal goal for this project.

I can also try this change out sometime if you'd prefer not to bother with it.

carbocation committed:
Fix go mod.

carbocation committed:
And mod.

carbocation committed:
Update buildBucketMultiple to support slices of images which do not
have the same size.

carbocation committed:
Swap the transparent color to be the first color in the palette.
This addresses a personal need and is also responsive to https://github.com/ericpauley/go-quantize/issues/3
